supply/lib/supply/client.rb

```diff
@@ -45,7 +45,11 @@ def self.service_account_authentication(params: nil)
     # Initializes the service and its auth_client using the specified information
     # @param service_account_json: The raw service account Json data
     def initialize(service_account_json: nil, params: nil)
-      auth_client = Google::Auth::ServiceAccountCredentials.make_creds(json_key_io: service_account_json, scope: self.class::SCOPE)
+      if params[:json_token] || params[:json_token_data]
+        auth_client = Google::Auth::UserRefreshCredentials.make_creds(json_key_io: service_account_json, scope: self.class::SCOPE)
+      else
+        auth_client = Google::Auth::ServiceAccountCredentials.make_creds(json_key_io: service_account_json, scope: self.class::SCOPE)
```

Comment:
This method is now failing for with `NoMethodError: [!] undefined method `gsub' for nil:NilClass `. Did this break for you?

Reply:
Good catch, I'd accidentally caused `service_account_authentication` to return `nil` when using a json key. Should be fixed now.